File: Java/Exercism/Hamming.java
public class Hamming {

    public String left = "";
    public String right = "";

    
    public Hamming(String leftStrand, String rightStrand) {
        
        if (leftStrand.length() != rightStrand.length()){
            throw new IllegalArgumentException("strands must be of equal length");
        }
        else {
            this.left = leftStrand;
            this.right = rightStrand;
        }
    }

    public int getHammingDistance() {

        int differenceCount = 0;
        
        for (int i = 0; i < this.left.length(); i++){
            if (this.left.charAt(i) != this.right.charAt(i)){
                differenceCount++;
            }
        }
        return differenceCount;
    }
}

File: JavaScript/Exercism/Hamming.js
//
// This is only a SKELETON file for the 'Hamming' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const compute = (Fham, Sham) => {
    let difference_count = 0;
  
    if (Fham.length != Sham.length){
       throw new Error("strands must be of equal length");
    }
    
    for (let i = 0; i <= Fham.length; i++){
      if (Fham[i] != Sham[i]){
        difference_count++
      }
    }
    return difference_count;
  };
  